### DISCUSSION TITLE
How to get the similarity score

### DISCUSSION THREAD

**Discussion by m3xw3ll** [aurelio-labs/semantic-router#252] [Q&A]
In all examples the similarity score is `None` after running
`rl('How is the weather today?')`

Can you please tell me how to show the actual similarity from my custom query to the specific route?

**Comment by octocat20** [comment]
That is expected on the simple call path in many releases.

rl("...") returns a RouteChoice from the matched Route. Route itself only sets name / function_call, so similarity_score stays None unless the layer fills it in afterward.

To get the score, use retrieve_multiple_routes:

```python
choices = rl.retrieve_multiple_routes("How is the weather today?")
print(choices[0].name, choices[0].similarity_score)
```

That path attaches the aggregated route score. On newer versions the plain rl("...") call may also populate similarity_score, but retrieve_multiple_routes is the reliable way to inspect it.
